src/EnergyPlus/Autosizing/Base.hh

```diff
@@ -0,0 +1,165 @@
+    CoolingWaterDesAirOutletHumRatSizing,
+    CoolingWaterDesWaterInletTempSizing,
+    HeatingAirflowUASizing,
+    HeatingWaterDesAirInletTempSizing,
+    Unknown
+};
+
+enum class AutoSizingResultType
+{
+    NoError,    // no errors found
+    ErrorType1, // sizing error
+    ErrorType2  // uninitialized sizing type
+};
+
+struct BaseSizer
+{
+
+    Real64 stdRhoAir = 0.0;
+
+    bool getCoilReportObject = false; // provides access to coil reporting
+    bool isNotInitialized = true;     // indicates initializeWithinEP was called
+    AutoSizingResultType errorType;
+    AutoSizingType sizingType = AutoSizingType::Unknown;
+    std::string sizingString = "";
+    Real64 originalValue = 0.0;
+    Real64 autoSizedValue = 0.0;
+    bool wasAutoSized = false;
+    bool hardSizeNoDesignRun = false;
+    bool sizingDesRunThisAirSys = false;
+    bool sizingDesRunThisZone = false;
+    bool sizingDesValueFromParent = false;
+    bool airLoopSysFlag = false;
+    bool oaSysFlag = false;
+    std::string compType = "";
+    std::string compName = "";
+
+    bool sysSizingRunDone = false;
+    bool zoneSizingRunDone = false;
+    int curSysNum = 0;
+    int curOASysNum = 0;
+    int curZoneEqNum = 0;
+    int curDuctType = 0;
+    int curTermUnitSizingNum = 0; // index in zone equipment vector - for single duct, IU, and PIU
+    int numPrimaryAirSys = 0;
+    int numSysSizInput = 0;
+    bool doSystemSizing = false;
+    int numZoneSizingInput = 0;
+    bool doZoneSizing = false;
+
+    // terminal units
+    bool termUnitSingDuct = false; // single duct terminal unit
+    bool termUnitPIU = false;      // powered induction unit
+    bool termUnitIU = false;       // induction terminal unit
+    bool zoneEqFanCoil = false;    // fan coil zone equipment
+    bool otherEqType = false;      // this covers the ELSE type switch
+
+    bool printWarningFlag = false;
+    std::string callingRoutine = "";
+    Array1D<DataSizing::SystemSizingInputData> sysSizingInputData;
+    Array1D<DataSizing::ZoneSizingInputData> zoneSizingInput;
+    Array1D<DataSizing::ZoneEqSizingData> unitarySysEqSizing;
+    Array1D<DataSizing::ZoneEqSizingData> oaSysEqSizing;
+    Array1D<DataSizing::ZoneEqSizingData> zoneEqSizing;
+    Array1D<DataAirLoop::OutsideAirSysProps> outsideAirSys;
+    Array1D<DataSizing::TermUnitSizingData> termUnitSizing;
+    Array1D<DataSizing::ZoneSizingData> finalZoneSizing;
+    Array1D<DataSizing::SystemSizingData> finalSysSizing;
+    std::vector<AirLoopHVACDOAS::AirLoopDOAS> airloopDOAS;
+
+    virtual void initializeWithinEP(EnergyPlusData &state,
+                                    std::string const &_compType,
+                                    std::string const &_compName,
+                                    bool const &_printWarningFlag,
+                                    std::string const &_callingRoutine);
+
+    virtual Real64 size(Real64 originalValue, bool &errorsFound) = 0;
+
+    static void clear_state();
+
+protected:
```

Comment:
I made a few functions `protected` so they can only be called from instances of the class or derived classes.  That way clients don't accidentally call these methods thinking they will do something else.